src/clib/pioc_support.c

```diff
@@ -148,35 +188,43 @@ void pio_log(int severity, const char *fmt, ...)
        many tabs before the message. */
     if (!severity)
     {
-        strcpy(ptr, ERROR_PREFIX);
+        strncpy(ptr, ERROR_PREFIX, (rem_len > 0) ? rem_len : 0);
```

Comment:
I don't object to a change to using strncpy. I agree that the message should be tuncated.

We cannot, obviously, make the code proof to future programming bugs. Keeping code to a minimum is a known way of reducing bugs.